app/src/main/java/org/curiouslearning/container/MainActivity.java

```diff
@@ -90,35 +91,39 @@ protected void onCreate(Bundle savedInstanceState) {
         utmPrefs = getSharedPreferences(UTM_PREFS_NAME, MODE_PRIVATE);
         isReferrerHandled = prefs.getBoolean(REFERRER_HANDLED_KEY, false);
         selectedLanguage = prefs.getString("selectedLanguage", "");
-        initialSlackAlertTime= AnalyticsUtils.getCurrentEpochTime();
+        initialSlackAlertTime = AnalyticsUtils.getCurrentEpochTime();
         homeViewModal = new HomeViewModal((Application) getApplicationContext(), this);
         cachePseudoId();
+        if (!isInternetConnected(getApplicationContext())) {
+            logStartedInOfflineMode();
+        }
         InstallReferrerManager.ReferrerCallback referrerCallback = new InstallReferrerManager.ReferrerCallback() {
             @Override
             public void onReferrerReceived(String deferredLang, String fullURL) {
                 String language = deferredLang.trim();
 
-                if (!isReferrerHandled ) {
+                if (!isReferrerHandled) {
                     SharedPreferences.Editor editor = prefs.edit();
                     editor.putBoolean(REFERRER_HANDLED_KEY, true);
                     editor.apply();
-                    if((language!=null && language.length()>0) || fullURL.contains("curiousreader://app")) {
-                        validLanguage(language, "google", fullURL.replace("deferred_deeplink=",""));
+                    if ((language != null && language.length() > 0) || fullURL.contains("curiousreader://app")) {
+                        validLanguage(language, "google", fullURL.replace("deferred_deeplink=", ""));
                         String pseudoId = prefs.getString("pseudoId", "");
                         String manifestVrsn = prefs.getString("manifestVersion", "");
-                        String lang ="";
-                        if(language!=null && language.length()>0)
-                            lang =  Character.toUpperCase(language.charAt(0))
+                        String lang = "";
```

Comment:
I hope you're doing well! I wanted to request if it would be possible to have the formatting changes in a separate pull request. It would make it a lot easier to review the code changes more effectively and ensure everything is aligned.

Thanks so much for your understanding, and let me know if you have any questions!

Reply:
Done Rajesh, I have already resolved most of the formatting issue!